.github/workflows/pull-request-check-rust-api.yaml

```diff
@@ -0,0 +1,105 @@
+name: Build and Test the Rust API
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    branches: [ develop ]
+env:
+  default_build_dir: "build/"
+  default_solver: "minisat2"
+
+# For now, we support two jobs: A Linux and a MacOS based one.
+# Both of the jobs use CMake, as we only support building the Rust
+# API with CMake (due to dependencies on CMake plugins).
+jobs:
+  check-ubuntu-22_04-cmake-clang-rust:
+    runs-on: ubuntu-22.04
+    env:
+      CC: "ccache /usr/bin/clang-13"
+      CXX: "ccache /usr/bin/clang++-13"
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Fetch dependencies
+        env:
+          # This is needed in addition to -yq to prevent apt-get from asking for
+          # user input
+          DEBIAN_FRONTEND: noninteractive
+        run: |
+          sudo apt-get update
+          sudo apt-get install --no-install-recommends -yq clang-13 clang++-13 gdb maven jq flex bison libxml2-utils cpanminus ccache z3
+          make -C src minisat2-download
+          cpanm Thread::Pool::Simple
+      - name: Confirm z3 solver is available and log the version installed
+        run: z3 --version
```

Comment:
Won't be used, so don't needs to be installed.

Reply:
Was thinking about this in particular that it could be used to test against in the future, but I suppose I'm not going to need it yet, so better to keep it clean.